app/src/org/commcare/android/database/global/models/ApplicationRecord.java

```diff
@@ -56,4 +84,104 @@ public int getStatus() {
     public void setStatus(int status) {
         this.status = status;
     }
+
+    public String getUniqueId() {
+        return uniqueId;
+    }
+
+    public void setUniqueId(String id) {
+        this.uniqueId = id;
+    }
+
+    public String getDisplayName() {
+        return this.displayName;
+    }
+
+    public void setDisplayName(String appName) {
+        this.displayName = appName;
+    }
+
+    public void setArchiveStatus(boolean b) {
+        this.isArchived = b;
+    }
+
+    public boolean isArchived() {
+        return this.isArchived;
+    }
+
+    @Override
+    public String toString() {
+        return this.displayName;
+    }
+
+    public void setResourcesStatus(boolean b) {
+        this.resourcesValidated = b;
+    }
+
+    public boolean resourcesValidated() {
+        return this.resourcesValidated;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord represents an app generated from
+     * a pre-Multiple Apps version of CommCare that does not have profile files
+     * with uniqueId and displayName
+     */
+    public boolean preMultipleAppsProfile() {
+        return this.preMultipleAppsProfile;
+    }
+    
+    public void setPreMultipleAppsProfile(boolean b) {
+        this.preMultipleAppsProfile = b;
+    }
+
+    /**
+     * Returns true if this ApplicationRecord was just generated from the a
+     * different ApplicationRecord format via the db upgrader, because it was 
+     * initially installed on a phone with a pre-Multiple Apps version of CommCare
+     */
+    public boolean convertedByDbUpgrader() {
+        return this.convertedViaDbUpgrader;
+    }
+
+    public void setConvertedByDbUpgrader(boolean b) {
+        this.convertedViaDbUpgrader = b;
+    }
+
+    /**
+     * Completes a full uninstall of the CommCare app that this ApplicationRecord represents
+     */
+    public void uninstall(Context c) {
+        CommCareApplication._().initializeAppResources(new CommCareApp(this));

```

Comment:
This is actually not a problem, at least not by all evidence thus far. There is currently no notion of unseating an app other than to seat a different app. Since CommCareApplication._().getCurrentApp() is the hook through which all parts of the app check what the currently-seated app is, calling initializeAppResources(CommCareApp), which overrides the currentApp field in CommCareApplication with the app passed in, is sufficient for both unseating the old and seating the new. Since the CommCareApp object is something that is recreated from the ApplicationRecord each time an app is seated, there isn't anything obvious to be done when "unseating" an app.

If you know of or suspect any weird behavior that this will cause, definitely tell me because if so, it will cause problems in many more places than just this. But from my understanding of how the CommCareApp/CommCareApplication code works this should be OK. And it's being done often and in many places, without hiccup thus far.